iac-extensions/spring-config/src/test/java/org/sonar/iac/springconfig/parser/properties/PropertiesParseTreeVisitorTest.java

```diff
@@ -0,0 +1,249 @@
+/*
+ * SonarQube IaC Plugin
+ * Copyright (C) 2021-2024 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.iac.springconfig.parser.properties;
+
+import java.util.stream.Stream;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+import org.junit.jupiter.params.provider.ValueSource;
+import org.sonar.iac.springconfig.tree.api.File;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.params.provider.Arguments.arguments;
+import static org.sonar.iac.springconfig.SpringConfigAssertions.assertThat;
+import static org.sonar.iac.springconfig.parser.properties.PropertiesTestUtils.createPropertiesFileContext;
+
+class PropertiesParseTreeVisitorTest {
+
+  @Test
+  void shouldReadSimpleProperty() {
+    var code = "database=h2";
+
+    var file = parseProperties(code);
+
+    var databaseKeyValue = file.profiles().get(0).properties().get(0);
+    assertThat(databaseKeyValue.key().value().value()).isEqualTo("database");
+    assertThat(databaseKeyValue.key().textRange()).hasRange(1, 0, 1, 7);
+    assertThat(databaseKeyValue.key().value().textRange()).hasRange(1, 0, 1, 7);
+
+    assertThat(databaseKeyValue.value().value().value()).isEqualTo("h2");
+    assertThat(databaseKeyValue.value().textRange()).hasRange(1, 9, 1, 10);
+    assertThat(databaseKeyValue.value().value().textRange()).hasRange(1, 9, 1, 10);
+  }
+
+  @Test
+  void shouldReadPropertyWithoutValue() {
+    var code = "foo";
+
+    var file = parseProperties(code);
+
+    var databaseKeyValue = file.profiles().get(0).properties().get(0);
+    assertThat(databaseKeyValue.key().value().value()).isEqualTo("foo");
+    assertThat(databaseKeyValue.key().textRange()).hasRange(1, 0, 1, 2);
+    assertThat(databaseKeyValue.key().value().textRange()).hasRange(1, 0, 1, 2);
+
+    assertThat(databaseKeyValue.value()).isNull();
+  }
+
+  @Test
+  void shouldReadSimpleProperties() {
+    var code = """
```

Comment:
Do we also want to test the colon as a delimiter?
Or a key with an array inside?

Reply:
It is already tested in `PropertiesParserBaseVisitorTest` but I'm ok to add it here as well as it is a test on different abstraction level.